### PR TITLE
fix(kinput): add show-masking-toggle prop [KHCP-13466]

### PR DESCRIPTION
**Pull request by portikM**
# Summary

Addresses: https://konghq.atlassian.net/browse/KHCP-13466

Add `showMaskingToggle` prop which allows toggling masking on and off when `type="password"`

Screenshot 2024-10-10 at 11 20 36 AM
Screenshot 2024-10-10 at 11 20 24 AM

### CODE REVIEW

**Review by Justineo — COMMENTED** [review]
We may want to turn off the built-in password reveal for Edge in case its toggle button overlaps with ours. (Edge has a strategy that only shows the toggle button if you start typing from nothing.)

image

https://learn.microsoft.com/en-us/microsoft-edge/web-platform/password-reveal#remove-the-password-reveal-control

**Comment by Justineo** [comment]
And long text is overlapping with the button.

image

File: src/components/KInput/KInput.vue [hunk 1]
```diff
@@ -35,18 +35,37 @@
...
+          @click.stop="maskValue = !maskValue"
+        >
+          <VisibilityIcon
+            v-if="maskValue"
+            decorative
+          />
+          <VisibilityOffIcon
+            v-else
+            decorative
+          />
```
Comment (Justineo) [inline src/components/KInput/KInput.vue:62]: Another question: should the icon reflect the current status or action? Current implementation is opposite to our login page's password input:

image

**Comment by portikM** [comment]
> We may want to turn off the built-in password reveal for Edge in case its toggle button overlaps with ours. (Edge has a strategy that only shows the toggle button if you start typing from nothing.)
> 
> image
> https://learn.microsoft.com/en-us/microsoft-edge/web-platform/password-reveal#remove-the-password-reveal-control

Fixed

File: src/components/KInput/KInput.vue (hunk 1 above)
Comment (portikM) [reply src/components/KInput/KInput.vue:62]: Oh good catch -- I looked at some other website to look up how they handle that but aligning with our login page makes more sense. Fixed.

**Review by Justineo — APPROVED** [review]
LGTM 👍

**Review by adamdehaven — APPROVED** [review]

**Pull request merged**